### ISSUE TITLE
5.2 and 5.3

### ISSUE THREAD

**Issue by eoftedal** [OWASP/owasp-masvs#11]
If you pin the public-key (5.3), is there really a need for a certificate signed by CA (5.2)? The CA-system aids a system/app/browser when accepting an unknown certificate from a host, but in this case the certificate is already known. If another host presented the same public key (and had the same private key), the security of that keypair would be compromised by this other host anyways.
So maybe 5.2 should be modified with something like "unless the public key is pinned".

**Comment by sushi2k** [comment]
I think the difference here is that 5.2 is a basic protection mechanism (level 1) and SSL Pinning is already level 2. Therefore a differentiation is already done.

**Comment by b-mueller** [comment]
Yes, on level 2 public key pinning overrides the CA requirement from level 1.

But, what about requirement 5.6:

"Verify that the app uses its own certificate store. Make sure that the CA certificate in the device trust store are ignored."

IMHO this a valid alternative to SSL pinning? If the app has its own store that only contains the CA certificate owned by the company. Isn't this actually a cleaner solution than SSL pinning (which is more of a hack)?

**Comment by sushi2k** [comment]
SSL Pinning is still a more secure approach. Remember DigiNotar? Just because you only have the CA would still make you vulnerable if the CA is compromised. If you Pin against the public key even a compromised CA would not affect your pinned connection. 
If you pin against the public key that is the cleanest solution, I think. Especially as once the certificate expires you can generate a new one with the old CSR and get a new certificate with the same public key, so no need to change the App as long as you have the CSR to generate a new certificate.

**Comment by jmanico** [comment]
If you only store only the one CA public key (ie: CA certificate pruning similar to https://codeascraft.com/2013/07/16/reducing-the-roots-of-some-evil/ ) you reduce the attack surface but do not eliminate MITM possibilities if that CA is rogue or its private signing key is stolen or your client side validation logic is flawed...

--
Jim Manico
@Manicode
Secure Coding Education
+1 (808) 652-3805

**Comment by b-mueller** [comment]
@sushi2k what I meant is that the company is running their own PKI instead of relying on a third-party CA. 
@jmanico true, SSL pinning reduces the attack surface even more, but it could also be argued that not trusting external CAs is _good enough_ for a L2 verification. As an example, an app I recently testing implemented their own CA store, which certainly makes MITM attacks very difficult. I'd have a hard time requiring them to also add SSL pinning, which is personally think is a rather hack-ish solution that comes with its own set of problems (e.g., how to handle certificate expiry, etc.). 
How about having an either/or requirement from L2?
